impress-org/givewp
# Donation History and Dashboard tabs are now dynamically populated

Resolves #5445 

## Description
This PR introduces the `donor-profile/donations` REST endpoint, which provides data needed to dynamically populate the Donation History and Dashboard tabs. Additionally, the PR improves on the Donation Table component, to enable pagination.

Note that there are various areas of UI polish that this PR leaves unfinished (styling donation rows exactly according to figma, formatting currencies, etc). This PR is focused on functionality, ahead of addressing aesthetic issues which will come in a future PR.

## Affects
This PR affects REST API endpoints, and donor-facing experiences with the Donor Profile app. Specifically, it ensures that the Dashboard and Donation History tabs are populated with data relevant to the currently logged in donor.

## Visuals
https://www.loom.com/share/3aa5f61bb52f4af587fb7f9b2cf5cff1

## Pre-review Checklist

## Testing Instructions

1. Create a new page
2. Add a Donor Profile block
3. Check that Dashboard content, and Donation History are correct.
